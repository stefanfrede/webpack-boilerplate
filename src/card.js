import { html } from 'lit-html';

import './css/index.css';
import imgsrc from './img/card-top.jpg';

// Define a template
const card = () =>
  html`
    <div class="container mx-auto py-8 flex flex-col">
      <div class="max-w-sm rounded overflow-hidden shadow-lg self-center">
        <img class="w-full" src="${imgsrc}" alt="Sunset in the mountains" />
        <div class="px-6 py-4">
          <div class="font-bold text-xl mb-2">The Coldest Sunset</div>
          <p class="text-gray-700 text-base">
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
            Voluptatibus quia, nulla! Maiores et perferendis eaque,
            exercitationem praesentium nihil.
          </p>
        </div>
        <div class="px-6 py-4">
          <span
            class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
            >#photography</span
          >
          <span
            class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
            >#travel</span
          >
          <span
            class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700"
            >#winter</span
          >
        </div>
      </div>
    </div>
  `;

export default card;
