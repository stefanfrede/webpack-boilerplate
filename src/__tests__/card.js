import card from '../card';

it('renders correctly', () => {
  expect(card()).toMatchSnapshot();
});
