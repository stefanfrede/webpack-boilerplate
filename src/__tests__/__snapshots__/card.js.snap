// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
TemplateResult {
  "processor": DefaultTemplateProcessor {},
  "strings": Array [
    "
    <div class=\\"container mx-auto py-8 flex flex-col\\">
      <div class=\\"max-w-sm rounded overflow-hidden shadow-lg self-center\\">
        <img class=\\"w-full\\" src=\\"",
    "\\" alt=\\"Sunset in the mountains\\" />
        <div class=\\"px-6 py-4\\">
          <div class=\\"font-bold text-xl mb-2\\">The Coldest Sunset</div>
          <p class=\\"text-gray-700 text-base\\">
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
            Voluptatibus quia, nulla! Maiores et perferendis eaque,
            exercitationem praesentium nihil.
          </p>
        </div>
        <div class=\\"px-6 py-4\\">
          <span
            class=\\"inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2\\"
            >#photography</span
          >
          <span
            class=\\"inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2\\"
            >#travel</span
          >
          <span
            class=\\"inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700\\"
            >#winter</span
          >
        </div>
      </div>
    </div>
  ",
  ],
  "type": "html",
  "values": Array [
    Object {
      "canInstrument": true,
      "createTransformer": [Function],
      "getCacheKey": [Function],
      "process": [Function],
    },
  ],
}
`;
